learningresources/tests/test_views.py

```diff
@@ -63,11 +59,82 @@ def test_create_repo_post(self):
     def test_listing_unauthorized(self):
         """View listing page."""
         # Not authorized to view this repository...
-        resp = self.client.get("/repositories/99/", follow=True)
-        body = resp.content.decode("utf-8")
-        self.assertTrue(resp.status_code == UNAUTHORIZED)
+        body = self.assert_status_code(
+            "/repositories/99/",
+            UNAUTHORIZED,
+            return_body=True
+        )
         self.assertTrue("unauthorized" in body)
 
+    def test_listing_importcourse_perms(self):
+        """
+        Tests the listing page with different user permissions
+        to check who can see the import course html
+        """
+        self.logout()
+        self.login(self.USERNAME_NO_REPO)
+        # user has no permissions at all
+        self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        # user has author permissions and cannot see the import for the repo
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_AUTHOR
+        )
+        body = self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            HTTP_OK,
+            return_body=True
+        )
+        self.assertFalse("Import Course</a>" in body)
+        # user has no permissions
+        remove_user_from_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_AUTHOR
+        )
+        self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        # user has curator permissions and can see the the import for the repo
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_CURATOR
+        )
+        body = self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            HTTP_OK,
+            return_body=True
+        )
+        self.assertTrue("Import Course</a>" in body)
```

Comment:
For tests which check for the ability to see the import page, a better check is to hit the import page and get denied, rather than hit the view page and look for text.

I know the display of that link is dependent upon the same permission setting that allows view of the page, but it still seems better to test the actual thing.